docs/developer-docs/smart-contracts/advanced-features/simd.mdx

```diff
@@ -0,0 +1,142 @@
+demanding top on-chain performance, such as Artificial Intelligence (AI),
+image processing (NFTs), games, scientific decentralized applications (dApps), and more.
+
+However, a significant performance boost is also possible for "classical"
+blockchain operations implemented in canisters. For example, reward distribution
+or cryptographic operations might benefit from the new WebAssembly SIMD instructions.
+
+### What is WebAssembly SIMD?
+
+WebAssembly SIMD (Single Instruction, Multiple Data) is a set of more than 200 deterministic
+vector instructions defined in the [WebAssembly core specification](https://webassembly.github.io/spec/core/syntax/instructions.html#vector-instructions). This parallel processing significantly accelerates specific tasks within canisters running on the ICP blockchain.
+
+The SIMD functionality is available on each node participating in the ICP network.
+
+### What Does WebAssembly SIMD Mean for Developers?
+
+WebAssembly development on the ICP blockchain reaches a new level of performance with SIMD support. Developers can now:
+
+* **Optimize code for computationally heavy tasks:** Identify areas within their canisters that can benefit from SIMD instructions and tailor their code for accelerated performance.
+* **Unlock new possibilities:** Explore novel functionalities and complex applications that were previously limited by processing power.
+* **Build a future-proof foundation:** ICP's embrace of cutting-edge technology like WebAssembly SIMD positions developers at the forefront of blockchain innovation.
+
+## Enabling WebAssembly SIMD
+
+There are two main ways to benefit from WebAssembly SIMD in a smart contract:
+
+1. **Loop auto-vectorization:** Just enabling the WebAssembly SIMD and recompiling the project might be enough to get a significant performance boost. This is usually simple, error-proof, and often a one-line change. This is often the recommended first step, but the result depends heavily on the used algorithms, libraries, and compilers.
+
+2. **SIMD intrinsics:** Some computation-heavy functions might be rewritten using direct SIMD instructions. This opens the full SIMD potential, but in many cases some core canister algorithms must be completely rewritten using new instructions.
+
+## Using Loop Auto-Vectorization
+
+To leverage the loop auto-vectorization, the WebAssembly SIMD instructions should be enabled globally for the entire workspace or locally for specific functions within the canister. Once the instructions are available to the compiler, it automatically converts some normal loops into loops with parallel computations.
+
+While the change is easy and error-proof, the result in practice depends on many factors, like the algorithm itself, the compiler optimization level and options, project dependencies, etc.
+
+### Enabling SIMD Example
+
+To enable WebAssembly SIMD instructions globally, for the whole workspace and all its dependencies:
+
+<AdornedTabs groupId="languages">
```

Comment:
Using tabs is not necessary if there is only one language being shown - are there any other language examples that will be added?

Reply:
AFAIK, no SIMD plans for Motoko. Maybe there will be something from JS/Python side...

Reply:
I'll try to merge now as is and see later if there will be more examples...

Reply:
Sounds good, thanks!